source/common/http/filter/fault_filter.cc

```diff
@@ -133,17 +124,22 @@ void FaultFilter::postDelayInjection() {
   // Delays can be followed by aborts
   if (config_->runtime().snapshot().featureEnabled("fault.http.abort.abort_percent",
                                                    config_->abortPercent())) {
-    Http::HeaderMapPtr response_headers{new HeaderMapImpl{
-        {Headers::get().Status, std::to_string(config_->runtime().snapshot().getInteger(
-                                    "fault.http.abort.http_status", config_->abortCode()))}}};
+    abortWithHTTPStatus();
     config_->stats().aborts_injected_.inc();
```

Comment:
The reason I left it out of that function was because in future, we might add `abortWithGRPCStatus` and `abortWithHTTP2Error` . Rather than duplicating the stats increment counter in all 3 functions, I thought its better to keep it here. What do you think?